app/src/main/java/ch/sdp/vibester/activity/DownloadActivity.kt

```diff
@@ -0,0 +1,114 @@
+package ch.sdp.vibester.activity
+
+import android.Manifest
+import android.app.DownloadManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Environment
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import ch.sdp.vibester.R
+import ch.sdp.vibester.api.ItunesMusicApi
+import ch.sdp.vibester.model.Song
+import okhttp3.OkHttpClient
+import org.w3c.dom.Text
+import java.lang.IllegalArgumentException
+
+class DownloadActivity : AppCompatActivity() {
```

Comment:
Should add documentations

Reply:
for all public methods of class (except override)

Reply:
Done!